src/trezor-connect-bridge.ts

```diff
@@ -0,0 +1,64 @@
+import TrezorConnect, { DEVICE_EVENT, DEVICE } from '@trezor/connect-web';
+import type {
+  Manifest,
+  ConnectSettings,
+  EthereumSignTransaction,
+  Params,
+  EthereumSignMessage,
+  EthereumSignTypedDataTypes,
+  EthereumSignTypedHash,
+} from '@trezor/connect-web';
+import type { TrezorBridge } from './trezor-bridge';
+
+export class TrezorConnectBridge implements TrezorBridge {
+  model?: string;
+
+  trezorConnectInitiated = false;
+
+  async init(
+    settings: {
+      manifest: Manifest;
+    } & Partial<ConnectSettings>,
+  ) {
+    TrezorConnect.on(DEVICE_EVENT, (event) => {
+      if (event.type !== DEVICE.CONNECT) {
```

Comment:
It's possible to force typescript type inference by using the event discriminator, that saves the use of `hasProperty` and avoid importing `@metamask/utils` as a dependency.